test/jasmine/tests/annotations_test.js

```diff
@@ -428,10 +431,12 @@ describe('annotations relayout', function() {
         {'annotations[100]': {text: 'bad index'}}
     ].forEach(function(update) {
         it('warns on ambiguous combinations and invalid values: ' + JSON.stringify(update), function() {
-            Plotly.relayout(gd, update);
-            expect(Loggers.warn).toHaveBeenCalled();
-            // we could test the results here, but they're ambiguous and/or undefined so why bother?
-            // the important thing is the developer is warned that something went wrong.
+            Plotly.relayout(gd, update)
+            .then(function() {
+                expect(Loggers.warn).toHaveBeenCalled();
+                // we could test the results here, but they're ambiguous and/or undefined so why bother?
+                // the important thing is the developer is warned that something went wrong.
+            });
```

Comment:
This one needs a `.catch(failTest).then(done)` now - which of course requires the `it` above to have `done` in its signature.

Reply:
OK looking down below there are a lot more in this category: once the test contains async code, it needs to be declared as such and finished with `.catch(failTest).then(done)`, otherwise none of the assertions in any `.then` blocks will actually be included in the test. They'll run, but failures won't cause the test to fail. Or perhaps would make it look like the next test failed? I don't quite recall but it's not fun 😅

Reply:
Fixed in https://github.com/plotly/plotly.js/pull/5340/commits/97574f966612ce1d6b669a35c10c356e09c8da7a.